doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1311 @@
+library context is required.
+
+=head3 Deprecated functions that should be changed if B<OSSL_LIB_CTX> is used
+
+All functions listed below with a I<NAME> have a replacment function I<NAME_ex>
+that takes B<OSSL_LIB_CTX> as an additional argument.
+
+=over 4
+
+=item EC_KEY_new_by_curve_name()
+
+=item EC_KEY_new()
+
+=item SRP_Calc_B()
+
+=item SRP_Calc_u()
+
+=item SRP_Calc_x()
+
+=item SRP_Calc_client_key()
+
+=item SRP_create_verifier_BN()
+
+=item SRP_create_verifier()
+
+=back
+
+=head3 New functions that use a B<OSSL_LIB_CTX>
+
+If a B<OSSL_LIB_CTX> is created then the following functions should set the
+created library context if required.
+
+=over 4
+
+=item EVP_ASYM_CIPHER_do_all_provided()
+
+=item EVP_ASYM_CIPHER_fetch()
+
+=item EVP_CIPHER_do_all_provided()
+
+=item EVP_CIPHER_fetch()
+
+=item EVP_default_properties_enable_fips()
+
+=item EVP_default_properties_is_fips_enabled()
+
+=item EVP_KDF_do_all_provided()
+
+=item EVP_KDF_fetch()
+
+=item EVP_KEM_do_all_provided()
+
+=item EVP_KEM_fetch()
+
+=item EVP_KEYEXCH_do_all_provided()
+
+=item EVP_KEYEXCH_fetch()
+
+=item EVP_KEYMGMT_do_all_provided()
+
+=item EVP_KEYMGMT_fetch()
+
+=item EVP_MAC_do_all_provided()
+
+=item EVP_MAC_fetch()
+
+=item EVP_MD_do_all_provided()
+
+=item EVP_MD_fetch()
+
+=item EVP_PKEY_CTX_new_from_pkey()
+
+=item EVP_RAND_do_all_provided()
+
+=item EVP_RAND()
+
+=item EVP_set_default_properties()
+
+=item EVP_SIGNATURE_do_all_provided()
+
+=item EVP_SIGNATURE_fetch()
+
+=item OSSL_CMP_CTX_new()
+
+=item OSSL_CMP_SRV_CTX_new()
+
+=item OSSL_CRMF_ENCRYPTEDVALUE_get1_encCert()
+
+=item OSSL_CRMF_MSG_create_popo()
+
+=item OSSL_CRMF_MSGS_verify_popo()
+
+=item OSSL_CRMF_pbm_new()
+
+=item OSSL_CRMF_pbmp_new()
+
+=item OSSL_DECODER_CTX_add_extra()
+
+=item OSSL_DECODER_CTX_new_for_pkey()
+
+=item OSSL_DECODER_do_all_provided()
+
+=item OSSL_DECODER_fetch()
+
+=item OSSL_ENCODER_CTX_add_extra()
+
+=item OSSL_ENCODER_do_all_provided()
+
+=item OSSL_ENCODER_fetch()
+
+=item OSSL_LIB_CTX_free()
+
+=item OSSL_LIB_CTX_load_config()
+
+=item OSSL_LIB_CTX_set0_default()
+
+=item OSSL_PROVIDER_add_builtin()
+
+=item OSSL_PROVIDER_available()
+
+=item OSSL_PROVIDER_do_all()
+
+=item OSSL_PROVIDER_load()
+
+=item OSSL_PROVIDER_set_default_search_path()
+
+=item OSSL_PROVIDER_try_load()
+
+=item OSSL_SELF_TEST_get_callback()
+
+=item OSSL_SELF_TEST_set_callback()
+
+=item OSSL_STORE_attach()
+
+=item OSSL_STORE_LOADER_do_all_provided()
+
+=item OSSL_STORE_LOADER_fetch()
+
+=item RAND_get0_primary()
+
+=item RAND_get0_private()
+
+=item RAND_get0_public()
+
+=item RAND_set_DRBG_type()
+
+=item RAND_set_seed_source_type()
+
+=back
+
+=head3 Behavioural Changes
+
+=head4 Key generation is slower.
+
+Key generation time has increased, especially for larger keys since at least
+64 rounds of the Miller-Rabin test are now also used for all prime generation,
+including RSA key generation.
+
+ * Reworked the treatment of EC EVP_PKEYs with the SM2 curve to
```

Comment:
This seems to be a separate topic that doesn't fit under the "Key generation is slower" heading

Reply:
I havent decided how to lay this out yet..

Reply:
reworked..